packages/ra-ui-materialui/src/list/FilterForm.js

```diff
@@ -127,36 +131,53 @@ export const mergeInitialValuesWithDefaultValues = ({
     initialValues,
     filters,
 }) => ({
-    initialValues: {
-        ...filters
-            .filter(
-                filterElement =>
-                    filterElement.props.alwaysOn &&
+    ...filters
+        .filter(
+            filterElement =>
+                filterElement.props.alwaysOn && filterElement.props.defaultValue
+        )
+        .reduce(
+            (acc, filterElement) =>
+                lodashSet(
+                    { ...acc },
+                    filterElement.props.source,
                     filterElement.props.defaultValue
-            )
-            .reduce(
-                (acc, filterElement) =>
-                    lodashSet(
-                        { ...acc },
-                        filterElement.props.source,
-                        filterElement.props.defaultValue
-                    ),
-                {}
-            ),
-        ...initialValues,
-    },
+                ),
+            {}
+        ),
+    ...initialValues,
 });
 
-const enhance = compose(
-    withStyles(styles),
-    withProps(mergeInitialValuesWithDefaultValues),
-    reduxForm({
-        form: 'filterForm',
-        enableReinitialize: true,
-        destroyOnUnmount: false, // do not destroy to preserve state across navigation
-        onChange: (values, dispatch, props) =>
-            props && props.setFilters(values),
-    })
-);
-
-export default enhance(FilterForm);
+const FormSpySubscription = { values: true, pristine: true };
+const EnhancedFilterForm = props => {
```

Comment:
please inverse the 2 components to make the file more readable

Reply:
bump

Reply:
This is weird, I inverted them already